test/core/test/injector-mock.test.ts

```diff
@@ -67,7 +72,1130 @@ test('correctly mocks namespaced', () => {
   `)).toMatchInlineSnapshot(`
     "const { vi } = await import('vitest')
     vi.mock('../src/add', () => {})
-    const AddModule = await import('../src/add')
-    const { default: add } = AddModule"
+    const __vi_import_0__ = await import('../src/add')"
+  `)
+})
+
+test('correctly access import', () => {
+  expect(hoistSimpleCode(`
+  import { vi } from 'vitest'
+  import add from '../src/add'
+  add();
+  vi.mock('../src/add', () => {})
+  `)).toMatchInlineSnapshot(`
+    "const { vi } = await import('vitest')
+    vi.mock('../src/add', () => {})
+    const __vi_import_0__ = await import('../src/add')
+
+      
+      
+      __vi_import_0__.default();"
   `)
 })
+
+describe('transform', () => {
+  const hoistSimpleCodeWithoutMocks = (code: string) => {
+    return hoistMocks(`import {vi} from "vitest";\nvi.mock(faker);\n${code}`, '/test.js', parse)?.code.trim()
+  }
+  test('default import', async () => {
+    expect(
+      await hoistSimpleCodeWithoutMocks(`import foo from 'vue';console.log(foo.bar)`),
+    ).toMatchInlineSnapshot(`
+      "const { vi } = await import('vitest')
+      vi.mock(faker)
+      const __vi_import_0__ = await import('vue')
+
+      ;
+      console.log(__vi_import_0__.default.bar)"
+    `)
+  })
+
+  test('named import', async () => {
+    expect(
+      await hoistSimpleCodeWithoutMocks(
+      `import { ref } from 'vue';function foo() { return ref(0) }`,
+      ),
+    ).toMatchInlineSnapshot(`
+      "const { vi } = await import('vitest')
+      vi.mock(faker)
+      const __vi_import_0__ = await import('vue')
+
+      ;
```

Comment:
I know that it was before your refactor, but this hanging `;` feels lonely, maybe we can remove it somehow?

Reply:
Ok

Reply:
This is an easy fix, I've already committed the code 😅